src/main/java/seedu/momentum/model/timer/Timer.java

```diff
@@ -0,0 +1,69 @@
+package seedu.momentum.model.timer;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+import seedu.momentum.model.project.Project;
+
+/**
+ * Represents a timer in momentum.
+ */
+public class Timer {
+    private Time startTime;
+    private Time stopTime;
+    private Project project;
```

Comment:
Do we really need to store the entire project object in a timer?

Reply:
The Timer class is only used to keep track of the active timers. Once the timer is stopped, the data is stored under a WorkDuration class (which has not reference to project), and put in a list inside the corresponding project.

I implemented it this way because I didn't want to store incomplete timers in a project, since it might cause problems if the timer wasn't stopped properly. Its also easier to check for active timers this way (see activeTimer in ModelManager)

And i think it would be easier to extend to having multiple concurrent timers if we decide to do that (then each timer needs to know which project its keeping track of)

Just storing the index won't work because the filtered list might change while the timer is running.

Reply:
Maybe we generate and use an unique project id?